File: makefile/src/main/java/name/kropp/intellij/makefile/psi/impl/MakefileFilenameMixin.java
package name.kropp.intellij.makefile.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiReference;
import com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReferenceSet;
import org.jetbrains.annotations.NotNull;

public class MakefileFilenameMixin extends ASTWrapperPsiElement {
    MakefileFilenameMixin(@NotNull ASTNode astNode) {
        super(astNode);
    }

    @NotNull
    @Override
    public PsiReference[] getReferences() {
        return new FileReferenceSet(getNode().getPsi()).getAllReferences();
    }
}

File: makefile/src/com/jetbrains/lang/makefile/psi/MakefileFilenameMixin.kt
package name.kropp.intellij.makefile.psi.impl

import com.intellij.extapi.psi.*
import com.intellij.lang.*
import com.intellij.psi.impl.source.resolve.reference.impl.providers.*

open class MakefileFilenameMixin internal constructor(astNode: ASTNode) : ASTWrapperPsiElement(astNode) {
  override fun getReferences(): Array<out FileReference> = FileReferenceSet(node.psi).allReferences
}
